### DISCUSSION TITLE
Isolated interconnect

### DISCUSSION THREAD

**Discussion by keurek** [NREL/ReEDS-2.0#38] [Q&A: Execution] [answered]
Is it possible to configure ReEDS to run as an isolated interconnect?

**Comment by keurek** [comment, marked as answer]
Yes, you can configure ReEDS as a single interconnect. For ReEDS v2020, ERCOT is the only interconnect that can be run in isolation by using "case_test.csv". Additional work is required to enable the western and eastern interconnections to be run in isolation.

---

The following three switches in "cases.csv" are interrelated and are used together to gather the regional information for a scenario.
Select all rows in "\inputs\regions\regions_{region_suffix}.csv" where the entry for column "{region_type}" is equal to "{GSw_region}."
* Example for the lower 48 US: Select all rows in "\inputs\regions\regions_default.csv" where the entry for column "country" is equal to "usa."
* Example for ERCOT: Select all rows in "\inputs\regions\regions_ercot.csv" where the entry for column "interconnect" is equal to "Texas." An example for ERCOT is shown in "cases_test.csv"

**regions_suffix [string]** – file pointer suffix for regional hierarchy file
path: \inputs\regions\regions_{regions_suffix}.csv
* default (for lower 48 US analysis)
* ercot (for ERCOT analysis)

**region_type [string]** – specify what region type will be used for "GSwS_region". Valid inputs are either column names in regions file or 'custom' (e.g. naris), where custom regions are defined in their own file. Model will not work for all regions.
* country (for lower 48 US analysis)
* interconnect (for ERCOT analysis)

**GSw_region [string]** – spatial extent of model regions
* usa (lower 48 US analysis)
* texas (for ERCOT analysis)

---

The default case configurations in "cases.csv" were designed for modeling the lower 48 United States. Therefore, there are possible issues with executing an interconnect in isolation, including but not limited to the following:

1. Natural gas prices are based on either national or census division supply curves. The natural gas prices are computed as a function of the quantity consumed relative to a reference quantity. Consuming less than the reference quantity drives the price downward; consuming more drives the price upward. When modeling a single interconnection, the user should either modify the reference gas quantity to account for a smaller spatial extent or use fixed gas prices in every census division (i.e., case configuration option GSw_GasCurve = 2). For example, if we execute ERCOT in isolation using census division supply curves, we may want to reduce the reference gas quantity for the West South Central (WSC) census division which includes Texas, Oklahoma, Arkansas, and Louisiana. Or we could assume the gas price in the WSC region is fixed.

1. Infeasibilities may arise in state-level constraints when only part of a state is represented in an interconnect. For example, the Western interconnection includes a small portion of Texas (El Paso). State-level constraints will be enforced for Texas, but El Paso may not be able to meet the requirement for all of Texas.

1. Certain constraints may not apply in every interconnect. Some examples include:
    * California State RPS REC trading constraints only apply to the West
    * CAIR and CSAPR only apply to certain states, so the emission limits may need to be adjusted
    * RGGI only applies to a subset of states in the northeast
    * California policies (e.g., SB32, California Storage Mandate) only apply to California

**Reply by astronobri** [reply]
I'm trying to run an isolated interconnect for Virginia specifically. I specified GSw_region = VA and region_type = state, using the default regions_default.csv. I turned off the GSw_AB32 switch since it's California only.

I'm getting an Augur error in C2_marginal_curtailment.py which says 

`ERROR:__main__:2022
ERROR:__main__:Traceback (most recent call last):
  File "Augur.py", line 177, in <module>
    ReEDS_Augur(next_year, prev_year, scen)
  File "Augur.py", line 101, in ReEDS_Augur
    mc_results, trans_region_curt = marginal_curtailment(curt_results)
  File "/gpfs/gpfs0/project/ceps-research/ReEDS_OpenAccess_bri/runs/regionstest_virginia_seq_RGGI/ReEDS_Augur/C2_marginal_curtailment.py", line 426, in marginal_curtailment
    net_load_adj_no_curt.values + exports_avail_path.values + imports_can_reduce.values))
ValueError: operands could not be broadcast together with shapes (8760,4) (8760,3) `

It's unhappy with the extra "p124" column in net_load_adj which looks to be all NaNs. I'm not really sure what's going on or why it's unhappy with p124 -- is there something else I'm missing for this case?

**Reply by vcelloho** [reply]
I'll look into getting someone who's worked more on the Augur development to explore this. One thing to test in the mean time is that the models isn't encountering other issues outside of Augur. I recommend a run with Augur disabled you can do so by changing the following case setting in the cases file.
GSw_SkipAugurYear = 2100

Changing the value to be after the last solve year will allow you to isolate this issue to Augur, and test performance with that feature disabled.

**Reply by wesleyjcole** [reply]
@astronobri Augur aggregates regions when doing capacity credit and curtailment calculations.  The regional aggregation is set by "capcredit_hierarchy_level" in the cases.csv file.  The default value of ccreg will include regions besides Virginia, which will create errors.  We've never net up ReEDS to run on a subset of an interconnect, so I don't have too many specific how-to's here, but it looks like you are making good progress.  And vcelloho's recommendation to test things with Augur turned off is also a good idea.

One issue you will run into as you use ReEDS in this down-scaled way is what to do about imports and exports in Virginia.  If you run Virginia as an isolated state, then all the decisions will be made assuming no interconnection capacity between Virginia and the rest of the Eastern Interconnection.  Given that, why not run the full Eastern Interconnection and just extract the Virginia results?

**Reply by astronobri** [reply]
Oh good point, I think running the full Eastern interconnection and just extracting the VA results is what I really want. I was over-complicating it! I'll give that a try. Thank you both!
